packages/react-scripts/scripts/start.js

```diff
@@ -11,6 +11,11 @@
 
 process.env.NODE_ENV = 'development';
 
+// Load environment variables from .env file. Surpress warnings using silent
+// if this file is missing. dotenv will never modify any environment variables
+// that have already been set.
+require('dotenv').config({silent: true});
```

Comment:
Note that absolute imports aren't officially supported in create react app as of now, so even if this environment variable might work, there is no guarantees of it continuing to be supported in the future. Therefore I wouldn't encourage using it until we add official support for absolute imports.

With this in mind, I think it's fine if the environment only works for REACT_APP_* variables – nothing else is officially supported right now.